frontend/packages/dev-console/src/components/formik-fields/DropdownField.tsx

```diff
@@ -6,12 +6,36 @@ import { FormGroup } from '@patternfly/react-core';
 import { DropdownFieldProps } from './field-types';
 import { getFieldId } from './field-utils';
 
-const DropdownField: React.FC<DropdownFieldProps> = ({ label, helpText, required, ...props }) => {
+const DropdownField: React.FC<DropdownFieldProps> = ({
+  label,
+  helpText,
+  required,
+  consumedItems,
+  handleDuplicateEntry,
+  ...props
+}) => {
   const [field, { touched, error }] = useField(props.name);
   const { setFieldValue, setFieldTouched } = useFormikContext<FormikValues>();
   const fieldId = getFieldId(props.name, 'dropdown');
   const isValid = !(touched && error);
   const errorMessage = !isValid ? error : '';
+
+  const checkDuplicateItemEntry = (items, fieldReference): boolean => {
+    let occurence = 0;
+    if (items && items.length > 0) {
+      items.forEach((element) => {
+        if (element === fieldReference.value) {
+          occurence++;
+        }
+      });
+    }
+    return !(occurence < 2);
+  };
+  const isDuplicateEntry = checkDuplicateItemEntry(consumedItems, field);
+  if(handleDuplicateEntry) {
+    handleDuplicateEntry(isDuplicateEntry);
+  }
+
```

Comment:
The intention behind addition of this check in the DropdownField component is to extend the functionality of the component so that it can handle usage in the multi column field, similar to how it has been used for revisions. This check is not specific to revisions, it can be used to warn of duplication of items in any similar scenario.